datalad/interface/rerun.py

```diff
@@ -40,6 +42,13 @@
 
 lgr = logging.getLogger('datalad.interface.rerun')
 
+rerun_assume_ready_opt = copy(assume_ready_opt)
+rerun_assume_ready_opt._doc += """
+Note that this option only affects inputs and outputs that were explicitly
+specified for the original run. Any additional outputs that are automatically
+inferred based on inspecting changed files in the run commit will still be
+prepared."""
```

Comment:
hm, would it be possible to make it also not bother unlocking "changed files"? It is what actually matters for any "incremental" use of `rerun` (as mentioned in https://github.com/datalad/datalad/issues/3206#issuecomment-781507078) where e.g. we get a new subject and just want to rerun prior "full dataset" analysis script which would be smart to run it only on new data, or just a script which just produces new outputs (e.g. `con/tineous` dumping new logs)

Reply:
In that case why wouldn't you just run the script as a new run command?

But anyway, if nobody chimes in to object, okay.

Reply:
I use `run` (and then `rerun`) to avoid the necessity to memoize and/or later re-create the entire invocation (I might have `--input` and even some `--output`s specified) with all the inputs and outputs.  `rerun` is a great helper in such cases, so I could just go to that dataset and `datalad rerun`

Reply:
My impression from reading that comment was that you were running the script without specifying inputs and outputs.   As for not recreating invocations, I personally would prefer those details to be kept in-tree, but we already know that I have different tastes than you regarding these things.

I'll switch the logic to include the automatic rerun outputs and change this docstring to note that they are included.